datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -3088,18 +3352,32 @@ mod tests {
                 child_properties,
                 expected_properties,
             } = self;
+
             let expected_properties =
                 expected_properties.expect("expected_properties not set");
-            let actual_properties =
-                calculate_union(child_properties, Arc::clone(&output_schema))
-                    .expect("failed to calculate union equivalence properties");
-            assert_eq_properties_same(
-                &actual_properties,
-                &expected_properties,
-                format!(
-                    "expected: {expected_properties:?}\nactual:  {actual_properties:?}"
-                ),
-            );
+
+            // try all permutations of the children
```

Comment:
I also updated the test runner to test with all child permutations (aka different orders) as the algorithm is not obviously symmetric